R/mergeTreeSE.R

```diff
@@ -0,0 +1,438 @@
+#' Merge TreeSE objects into single TreeSE object.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object or a list of 
+#' \code{\link{SummarizedExperiment}} objects.
+#' 
+#' @param y a \code{\link{SummarizedExperiment}} object 
+#' 
+#' @param assay_name A single character value for selecting the
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#' to be merged. (By default: \code{assay_name = "counts"})
+#' 
+#' @param join A single character value for selecting the joining method.
+#' Must be 'full', 'inner', 'left', or 'right'. 'left' and 'right' are disabled
+#' when more than two objects are being merged.  (By default: \code{join = "full"})
+#' 
+#' @param missing_values NA, 0, or a single character values specifying the notation
+#' of missing values. (By default: \code{missing_values = 0})
+#' 
+#' @param verbose A single boolean value to choose whether to show messages. 
+#' (By default: \code{verbose = TRUE})
+#'
+#' @param ... optional arguments (not used).
+#'
+#' @return A single \code{TreeSummarizedExperiment} object.
+#'
+#' @details
+#' This function merges multiple \code{SummarizedExperiment} objects. It combines
```

Comment:
Perhaps OK this way. But makes me wonder whether it would be better to require that the users make conversions between types explicitly before applying this? So that only similar types could be merged. This could potentially help to avoid confusion. On the other hand it is handy that anything can be merged. But I am not sure if it is so common to merge different types that making the transformations _before_ applying the merge function would be a big deal to do by those who really need it.

At the minimum, there should be a warning if the user merges different types (if not already).

Reply:
How we should tackle this? Should we add a warning?

Reply:
I added warning